frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx

```diff
@@ -53,67 +57,92 @@ function InviteRow({ index }: { index: number }): JSX.Element {
+    const { appendInviteRow, resetInviteRows, inviteTeamMembers } = useActions(bulkInviteLogic)
 
     useEffect(() => {
-        if (invitedTeamMembers.invites.length) {
+        if (invitedTeamMembers.length) {
             onClose()
         }
     }, [invitedTeamMembers])
 
     return (
-        <>
-            <Modal
-                title={`Invite your team members${user?.organization ? ' to ' + user?.organization?.name : ''}`}
-                visible={visible}
-                onCancel={() => {
-                    resetInvites()
-                    onClose()
-                }}
-                onOk={inviteTeamMembers}
-                okText="Invite team members"
-                destroyOnClose
-                okButtonProps={{ disabled: !canSubmit, loading: invitedTeamMembersLoading }}
-                cancelButtonProps={{ disabled: invitedTeamMembersLoading }}
-                closable={!invitedTeamMembersLoading}
-            >
-                <div className="bulk-invite-modal">
-                    <div>
-                        Invite as many team members as you want. <b>Names are optional</b>, but it will speed up the
-                        process for your teammates.
-                    </div>
-                    <Row gutter={16} className="mt">
-                        <Col xs={12}>
-                            <b>Email (required)</b>
-                        </Col>
-                        <Col xs={12}>
-                            <b>First Name</b>
-                        </Col>
-                    </Row>
+        <Modal
+            title={`Inviting team members${user?.organization ? ' to ' + user?.organization?.name : ''}`}
+            visible={visible}
+            onCancel={() => {
+                resetInviteRows()
+                onClose()
+            }}
+            onOk={inviteTeamMembers}
+            okText={`Invite ${pluralize(
+                invites.filter((invite) => invite.isValid && invite.target_email).length,
+                'team member'
+            )}`}
+            destroyOnClose
+            okButtonProps={{ disabled: !canSubmit, loading: invitedTeamMembersLoading }}
+            cancelButtonProps={{ disabled: invitedTeamMembersLoading }}
+            closable={!invitedTeamMembersLoading}
+        >
+            <div className="bulk-invite-modal">
+                <p>
+                    An invite is <b>specific to an email address</b> and <b>expires after 3 days</b>.
+                    <br />
+                    Name can optionally be provided for the invitee's convenience.
+                </p>
+                <Row gutter={16}>
+                    <Col xs={11}>
+                        <b>
+                            Email address <i>(required)</i>
+                        </b>
+                    </Col>
+                    <Col xs={11}>
+                        <b>
+                            Name <i>(optional)</i>
```

Comment:
Hm, I agree this should be clearer. IMO though everyone should have _full_ names. In a professional setting it gets odd and confusing otherwise (even PostHog at 5 people ended up with 2 Jameses). I don't know of a B2B product where users have first names only.

Reply:
Hmmm you might be right, I think this is fine then.